libraries/AP_Airspeed/AP_Airspeed.cpp

```diff
@@ -138,175 +214,215 @@ AP_Airspeed::AP_Airspeed()
+        return;
+    }
+    
     // consider calibration complete when we have at least 15 samples
     // over at least 1 second
-    if (AP_HAL::millis() - _cal.start_ms >= 1000 &&
-        _cal.read_count > 15) {
-        if (_cal.count == 0) {
-            gcs().send_text(MAV_SEVERITY_INFO, "Airspeed sensor unhealthy");
+    if (AP_HAL::millis() - state[i].cal.start_ms >= 1000 &&
+        state[i].cal.read_count > 15) {
+        if (state[i].cal.count == 0) {
+            gcs().send_text(MAV_SEVERITY_INFO, "Airspeed[%u] sensor unhealthy", i);
         } else {
-            gcs().send_text(MAV_SEVERITY_INFO, "Airspeed sensor calibrated");
-            _offset.set_and_save(_cal.sum / _cal.count);
+            gcs().send_text(MAV_SEVERITY_INFO, "Airspeed[%u] sensor calibrated", i);
+            param[i].offset.set_and_save(state[i].cal.sum / state[i].cal.count);
         }
-        _cal.start_ms = 0;
+        state[i].cal.start_ms = 0;
         return;
     }
     // we discard the first 5 samples
-    if (_healthy && _cal.read_count > 5) {
-        _cal.sum += raw_pressure;
-        _cal.count++;
+    if (state[i].healthy && state[i].cal.read_count > 5) {
+        state[i].cal.sum += raw_pressure;
+        state[i].cal.count++;
     }
-    _cal.read_count++;
+    state[i].cal.read_count++;
 }
 
-// read the airspeed sensor
-void AP_Airspeed::read(void)
+// read one airspeed sensor
+void AP_Airspeed::read(uint8_t i)
 {
     float airspeed_pressure;
-    if (!enabled()) {
+    if (!enabled(i) || !sensor[i]) {
         return;
     }
-    float raw_pressure = get_pressure();
-    if (_cal.start_ms != 0) {
-        update_calibration(raw_pressure);
+    float raw_pressure = get_pressure(i);
+    if (state[i].cal.start_ms != 0) {
+        update_calibration(i, raw_pressure);
     }
     
-    airspeed_pressure = raw_pressure - _offset;
+    airspeed_pressure = raw_pressure - param[i].offset;
 
     // remember raw pressure for logging
-    _corrected_pressure = airspeed_pressure;
+    state[i].corrected_pressure = airspeed_pressure;
 
     // filter before clamping positive
-    _filtered_pressure      = 0.7f * _filtered_pressure  +  0.3f * airspeed_pressure;
+    state[i].filtered_pressure = 0.7f * state[i].filtered_pressure + 0.3f * airspeed_pressure;
 
     /*
       we support different pitot tube setups so user can choose if
       they want to be able to detect pressure on the static port
      */
-    switch ((enum pitot_tube_order)_tube_order.get()) {
+    switch ((enum pitot_tube_order)param[i].tube_order.get()) {
     case PITOT_TUBE_ORDER_NEGATIVE:
-        _last_pressure          = -airspeed_pressure;
-        _raw_airspeed           = sqrtf(MAX(-airspeed_pressure, 0) * _ratio);
-        _airspeed               = sqrtf(MAX(-_filtered_pressure, 0) * _ratio);
+        state[i].last_pressure  = -airspeed_pressure;
+        state[i].raw_airspeed   = sqrtf(MAX(-airspeed_pressure, 0) * param[i].ratio);
```

Comment:
i think its clearer the way it is (only some lines are in common)

Reply:
I'm fine either way, what I was observing though is that the raw_airspeed line is actually common, and extracting it would make it lest prone to copy paste errors (although I can't really think of any other tube orientations that make sense here :) )